### ISSUE TITLE
[Doc] Document Hazelcast k8s

### ISSUE THREAD

**Issue by hupling** [it-at-m/refarch#179]
### Bug description

Der Pod crashed, wenn man das Profil hazelcast-k8s und wenn man alle Properties angibt:
- HAZELCAST_OPENSHIFTSERVICENAME
- APP_SPRING-SESSION-HAZELCAST_NAMESPACE:

2024.09.27 10:59:33.802 | refarch-gateway | TraceId: | SpanId: ] | INFO | [main] | DiscoveryService | [StandardLoggerFactory.java : 56] - [10.138.32.88]:5701 [session_replication_group] [5.4.0] Kubernetes Discovery properties: { service-dns: null, service-dns-timeout: 5, service-name: a-refarch-gateway, service-port: 0, service-label: null, service-label-value: true, namespace: spseval, pod-label: null, pod-label-value: null, resolve-not-ready-addresses: true, expose-externally-mode: AUTO, use-node-name-as-external-address: false, service-per-pod-label: null, service-per-pod-label-value: null, kubernetes-api-retries: 3, kubernetes-master: https://kubernetes.default.svc}
2024.09.27 11:00:32.625 | refarch-gateway | TraceId: | SpanId: ] | WARN | [main] | AnnotationConfigReactiveWebServerApplicationContext | [AbstractApplicationContext.java : 632] - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'webSessionManager' defined in class path resource [org/springframework/session/config/annotation/web/server/SpringWebSessionConfiguration.class]: Unsatisfied dependency expressed through method 'webSessionManager' parameter 0: Error creating bean with name 'reactiveSessionRepository' defined in class path resource [de/muenchen/refarch/gateway/configuration/WebSessionConfiguration.class]: Unsatisfied dependency expressed through method 'reactiveSessionRepository' parameter 0: Error creating bean with name 'hazelcastInstance' defined in class path resource [de/muenchen/refarch/gateway/configuration/WebSessionConfiguration.class]: Failed to instantiate [com...

Wenn man HAZELCAST_OPENSHIFTSERVICENAME nicht angibt geht es. 

Ich hatte die Vermutung, dass es passiert, weil alle Request auf  / auf das Frontend weitergeleitet werden, dass ist, aber nicht der Fall

```
    - name: SPRING_CLOUD_GATEWAY_ROUTES_2_URI
      value: "http://a-frontend:8080/"
    - name: SPRING_CLOUD_GATEWAY_ROUTES_2_PREDICATES_0
      value: Path=/**
    - name: SPRING_CLOUD_GATEWAY_ROUTES_2_FILTERS_0
      value: RewritePath=/(?<urlsegments>.*), /$\{urlsegments}
    - name: SPRING_CLOUD_GATEWAY_ROUTES_2_FILTERS_1
      value: RemoveResponseHeader=WWW-Authenticate
```

### Expected behaviour

Hazelcast funktioniert wie beim bisherigen Gateway

### Steps for reproduction

siehe oben 

### Affected version

akutelle

### Affected component

refarch-gateway

### No duplicate

- [X] I confirm that this issue is not a duplicate

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by devtobi** [comment]
@simonhir Could you do further investigation on this topic?

**Comment by simonhir** [comment]
@hupling could you please provide the full error log and what values you set for the mentioned env vars?

**Comment by hupling** [comment]
so startet der Pod
```
    - name: APP_SPRING-SESSION-HAZELCAST_NAMESPACE
      value: 'spseval'
    # - name: HAZELCAST_OPENSHIFTSERVICENAME
    #   value: 'a-backend'
```
so stürtzt er ab
```
    - name: APP_SPRING-SESSION-HAZELCAST_NAMESPACE
      value: 'spseval'
     - name: HAZELCAST_OPENSHIFTSERVICENAME
       value: 'a-backend'
```

@simonhir ich habe dir Zugriff zum Namespace gegeben, dann kannst du dir selbstständig die Logs anschauen

**Comment by simonhir** [comment]
The following two warnings are the interesting ones in my opinion:

```
Hazelcast is starting in a Java modular environment (Java 9 and newer) but without proper access to required Java packages. Use additional Java arguments to provide Hazelcast access to Java internal API. The internal API access is used to get the best performance results. Arguments to be used:
 --add-modules java.se --add-exports java.base/jdk.internal.ref=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.management/sun.management=ALL-UNNAMED --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED
```

See internal https://stapelueberlauf.muenchen.de/index.php?qa=261&qa_1=hazelcast-starting-modular-environment-required-packages&show=261#q261 and https://docs.hazelcast.org/docs/3.12/manual/html-single/index.html#running-in-modular-java for fix.

```
Kubernetes API access is forbidden! Starting standalone. To use Hazelcast Kubernetes discovery, configure the required RBAC. For 'default' service account in 'default' namespace execute `kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast/master/kubernetes-rbac.yaml` If you want to use a different service account and a different namespace, you can update the mentioned rbac.yaml file accordingly and use it. Error Kubernetes API Cause details:
```

In my opinion this i no bug of the gateway and just requires some deployment config. What can be thought about is modifying the helm chart.

**Comment by simonhir** [comment]
For the second error `hazelcast.openshift-service-name` was missing. The permissions are [internally set by default](https://git.muenchen.de/openshift/projects/-/tree/main/lhm_refarch?ref_type=heads) as role `Default-View` is assigned to `default` service account.

**Comment by hupling** [comment]
hat vor der OpenSource Stellung auch nicht funktioniert

**Comment by simonhir** [comment]
> For the second error hazelcast.openshift-service-name was missing

Also requires deployment and service to expose port 5701

**Comment by simonhir** [comment]
To conclude: everything works as intended from the gateway-image side.
It requires configuration of the hazelcast vars and the port in deployment and service.

To make this more intuitive the following needs to be done:
- Add hazelcast vars to gateway README.md
- Document additional java options for hazelcast module warning in README.md
- Add ports to helm template

**Comment by hupling** [comment]
Reason: java.lang.IllegalStateException: No setter found for property: service-name

@simonhir ihr habt da noch einen fehler. keine getter/setter oder public

**Comment by simonhir** [comment]
Fixed with #210
